R-package/src/cmake/modules/FindLibR.cmake

```diff
@@ -0,0 +1,224 @@
+# CMake module used to find the location of R's
+# dll and header files.
+#
+# Borrows heavily from xgboost's R package:
+#
+# * https://github.com/dmlc/xgboost/blob/master/cmake/modules/FindLibR.cmake
+#
+# Defines the following:
+#  LIBR_FOUND
+#  LIBR_HOME
+#  LIBR_EXECUTABLE
+#  LIBR_INCLUDE_DIRS
+#  LIBR_LIB_DIR
+#  LIBR_CORE_LIBRARY
+# and a cmake function to create R.lib for MSVC
+
+if(NOT R_ARCH)
+  if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "4")
+    message(FATAL_ERROR "LightGBM's R package currently only supports 64-bit operating systems")
+  endif()
+  set(R_ARCH "x64")
+endif()
```

Comment:
```suggestion
if(NOT R_ARCH)
  if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "4")
    set(R_ARCH "i386")
  else()
    set(R_ARCH "x64")
  endif()
endif()

if(NOT ("${R_ARCH}" STREQUAL "x64"))
  message(FATAL_ERROR "LightGBM's R package currently only supports 64-bit operating systems")
endif()
```

Reply:
@StrikerRUS check it out! In case you haven't seen, GitHub now supports multi-line suggestions 😀

Reply:
Yeah, cool feature!